distro/modules/core.nix

```diff
@@ -1,7 +1,49 @@
 { config, lib, pkgs, ... }:
 let
-  kernel = builtins.path { path = builtins.getEnv "NIXOS_KERNEL"; };
-  stage-1-init = builtins.path { path = builtins.getEnv "NIXOS_STAGE_1_INIT"; };
+  stage-1-init = pkgs.writeShellScript "stage-1-init" ''
+    #!/bin/sh
+    # SPDX-License-Identifier: MPL-2.0
+
+    NEW_ROOT=""
+    NEW_INIT=""
+    BREAK=""
+    ARGS=""
+
+    for arg in "$@"; do
+      case "$arg" in
+        root=*)
+          NEW_ROOT=''${arg#root=}
+          ;;
+        init=*)
+          NEW_INIT=''${arg#init=}
+          ;;
+        rd.break=*)
+          BREAK=''${arg#rd.break=}
+          ;;
+        *)
+          ARGS="$ARGS $arg"
+          ;;
+      esac
+    done
+
+    if [ "$BREAK" = "1" ]; then
+      echo "Breaking into initramfs shell..."
+      exec /bin/sh
+    fi
+
+    if [ -z "$NEW_ROOT" ] || [ -z "$NEW_INIT" ]; then
+      echo "Error: 'root=' and 'init=' parameters are required."
+      exit 1
+    fi
+
+    mkdir /sysroot
+    mount -t ext2 "$NEW_ROOT" /sysroot
+    mount -t proc none /sysroot/proc
```

Comment:
I am a bit of confused here: `/sysroot/proc` is non-existential. Doesn't `mount` require the target directory to be created first?